File: app/models/trabajo_persona.py
from sqlalchemy import ForeignKey
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class TrabajoPersona(Base):
    __tablename__ = "trabajo_persona"

    id: Mapped[int] = mapped_column(primary_key=True)
    id_trabajo: Mapped[int] = mapped_column(ForeignKey("trabajo_grado.id_trabajo"))
    id_persona: Mapped[int] = mapped_column(ForeignKey("persona.id_persona"))
    id_rol: Mapped[int] = mapped_column(ForeignKey("rol.id_rol"))

File: flutter_app/lib/models/trabajo_persona.dart
class TrabajoPersona {
  final int id;
  final int idTrabajo;
  final int idPersona;
  final int idRol;

  TrabajoPersona({
    required this.id,
    required this.idTrabajo,
    required this.idPersona,
    required this.idRol,
  });

  factory TrabajoPersona.fromJson(Map<String, dynamic> json) {
    return TrabajoPersona(
      id: json['id'] as int,
      idTrabajo: json['id_trabajo'] as int,
      idPersona: json['id_persona'] as int,
      idRol: json['id_rol'] as int,
    );
  }
}
